### PR TITLE
Adding userstore group id support for group related operations

### PR DESCRIPTION
**Pull request by somindatommy**
Related to the 1st task of wso2/product-is#7913
Related to https://github.com/wso2/product-is/issues/7915

### Proposed changes
Calling UM methods for group-related operations.

**Related PRs:** [1] and [2].

Before merging the PR.
- [x] Merge PR [1]
- [x] Bump kernel version and framework in product-is
- [ ] Bump kernel version and framework in scim

[1] - https://github.com/wso2/carbon-identity-framework/pull/3659
[2] - https://github.com/wso2/carbon-kernel/pull/3042

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java [hunk 1]
```diff
@@ -564,4 +564,145 @@ public static String getCustomSchemaURI() {
...
+        } else {
+            return delimiter + attributeValue + delimiter;
+        }
+    }
+
+    /**
+     * Check whether the filter attribute support filtering with the domain embedded in the attribute value.
+     *
+     * @param attributeName Attribute to filter.
+     * @return True if the given attribute support embedding domain in attribute value..
```
Comment (bhagyasakalanka) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java:630]: Formatting error. Remove the extra full stops.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/SQLQueries.java [hunk 2]
```diff
@@ -28,6 +28,8 @@
     public static final String GET_ATTRIBUTES_SQL =
             "SELECT ATTR_NAME, ATTR_VALUE FROM IDN_SCIM_GROUP WHERE IDN_SCIM_GROUP.TENANT_ID=? AND " +
                     "IDN_SCIM_GROUP.ROLE_NAME=?";
+    public static final String GET_GROUP_ID_BY_NAME_SQL = "SELECT ATTR_VALUE FROM IDN_SCIM_GROUP WHERE TENANT_ID=? " +
+            "AND ROLE_NAME=? AND IDN_SCIM_GROUP.ATTR_NAME=?";
```
Comment (bhagyasakalanka) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/SQLQueries.java:32]: Do we need to add `IDN_SCIM_GROUP.ATTR_NAME`.  We can directly use `ATTR_NAME`

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+            } catch (IdentitySCIMException e) {
+                throw new UserStoreException(String.format("Error occurred while getting the group id of " +
+                        "group: %s in tenant: %s", group.getGroupName(), tenantId), e);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean getGroupIdByName(String groupName, Group group, UserStoreManager userStoreManager)
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java:145]: shouldn't we return the groupID here as the method name implies so.

Comment (somindatommy) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java:145]: This is a listener to communicate from SCIM to the kernel. By default, we return a boolean value for all listener methods. To preserve that I had added return type as `boolean` how we pass the group id is using the `group` param.

**Review by emswbandara — APPROVED** [review]

**Pull request merged**